Add a section for service load balancer cleanup (#14496)
kubernetes/website: content/en/docs/tasks/access-application-cluster/create-external-load-balancer.md

{{% capture discussion %}}

## Load Balancer cleanup

In usual case, the correlating load balancer resources in cloud provider should
be cleaned up soon after a LoadBalancer type service is deleted. But it is known
that there are various corner cases where resources can be leaved orphaned after
a service is deleted. Finalizer Protection for Service LoadBalancers was
introduced to prevent this from happening. By using finalizer, this feature
makes sure service resource is not fully deleted until the correlating load
balancer resources are deleted.

> Specifically, a finalizer with name `"service.kubernetes.io/load-balancer-cleanup"`

Reviewer: @MrHohn do you think this could stand being written in the active voice?

You could switch to something like:
> Specifically, if a Service has a load balancer resource created for it, _component X_
> will attach a finalizer named `"service.kubernetes.io/load-balancer-cleanup"`.

I assume _component X_ will be a controller; I haven't looked up which one.

The finalizer will only be removed after the load balancer resource is cleaned up.

> Above ensures resource won't be leaked even in corner cases.

Reviewer 2: I prefer the term "dangling" instead of "leaked in this scenario:
```
This prevents dangling load balancer resources even in corner cases such as the controller manager crashing. 
```

Author: Thanks, this is better.

Above ensures resource won't be leaked even in corner cases.

> This feature was introduced as Alpha in kubernetes v1.15. It can be enabled via

Reviewer: ```suggestion
This feature was introduced as alpha in Kubernetes v1.15. You can start using it by
```

(Kubernetes in upper case as per style guide; remainder is just my suggestion)

> feature gate `ServiceLoadBalancerFinalizer=true`.

Reviewer: ```suggestion
enabling the [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) `ServiceLoadBalancerFinalizer`.
```

{{% capture discussion %}}

> ## Load Balancer cleanup

Author: Picked the garbage collecting one :)

In usual case, the correlating load balancer resources in cloud provider should
be cleaned up soon after a LoadBalancer type Service is deleted. But it is known

> that there are various corner cases where resources can be leaved orphaned after

Author: Thanks, applied.

a Service is deleted. Finalizer Protection for Service LoadBalancers was

> introduced to prevent this from happening. By using finalizer, this feature

Author: Applied.

makes sure Service resource is not fully deleted until the correlating load
balancer resources are deleted.

> Specifically, if a Service has a load balancer resource created for it, service

Author: Good point to mention type=LoadBalancer. Applied.
